ext/pgsql/pgsql.c

```diff
@@ -3879,16 +3879,18 @@ static void php_pgsql_do_async(INTERNAL_FUNCTION_PARAMETERS, int entry_type)
 	switch(entry_type) {
 		case PHP_PG_ASYNC_IS_BUSY:
 			PQconsumeInput(pgsql);
-			RETVAL_LONG(PQisBusy(pgsql));
+			RETVAL_BOOL(PQisBusy(pgsql));
```

Comment:
shouldn't it be `RETVAL_BOOL(PqisBusy(pgsql) == 0);` ?

Reply:
It only ever returns 1 or 0 so this a one to one map to a bool, no?

Reply:
fair enough before I looked up I thought you had to give an actual boolean, yes it works like this.